virttest/qemu_vm.py

```diff
@@ -1442,11 +1432,15 @@ def sort_key(dev):
                                           params.get('strict_mode'),
                                           params.get(
                                               'workaround_qemu_qmp_crash'),
-                                          params.get('allow_hotplugged_vm'),
-                                          vm._caps)
+                                          params.get('allow_hotplugged_vm'))
         StrDev = qdevices.QStringDevice
         QDevice = qdevices.QDevice
 
+        # update capabilities by users willingness
+        if (params.get("qemu_force_use_drive_expression", "no") == "yes" and
+                Flags.BLOCKDEV in devices.caps):
+            devices.caps.clear_flag(Flags.BLOCKDEV)
```

Comment:
So you went the global way of overrides. I still believe it'd be better to support `qemu_force_use_drive_expression` in `images_define_by_params` to allow defining some drives with `-drive` together with other disks using `-blockdev`. It should be fairly straight-forward, similar to using a different HBA. Instead of removing the capability we'd have to check whether we want to use it and then whether we support it.

Reply:
Btw I don't even thing it'd be necessary to includ it into the `images_define_by_variables`, unless someone asks for it.

Reply:
I think that `qemu_force_use_drive_expression` should be a global switch for controlling the expression of block devices because we don't usually have any mixed-up cases, therefore, I'd suggest not bring it to `images_define_by_params` and `images_define_by_variables`.

About the forced values, IMO, we can introduce another attribute to the capabilities class to separate from the natural ones, but firstly let's keep it simple.